crates/lint/src/sol/info.rs

```diff
@@ -0,0 +1,191 @@
+pub fn is_mixed_case(s: &str) -> bool {
+    if s.len() <= 1 {
+        return true;
+    }
+
+    let re = Regex::new(r"^[a-z_][a-zA-Z0-9]*$").unwrap();
+    re.is_match(s)
+}
+
+/// Check if a string is PascalCase
+pub fn is_pascal_case(s: &str) -> bool {
+    if s.len() <= 1 {
+        return true;
+    }
+
+    let re = Regex::new(r"^[A-Z][a-z]+(?:[A-Z][a-z]+)*$").unwrap();
+    re.is_match(s)
+}
+
+/// Check if a string is SCREAMING_SNAKE_CASE, where
+/// numbers must always be preceded by an underscode.
+pub fn is_screaming_snake_case(s: &str) -> bool {
+    if s.len() <= 1 {
+        return true;
+    }
+
+    let re = Regex::new(r"^[A-Z_][A-Z0-9_]*$").unwrap();
+    let invalid_re = Regex::new(r"[A-Z][0-9]").unwrap();
+    re.is_match(s) && !invalid_re.is_match(s)
+}
+
+#[cfg(test)]
+mod test {
+    use std::path::Path;
+
+    use super::*;
+    use crate::{
+        linter::Lint,
+        sol::{SolidityLinter, FUNCTION_MIXED_CASE},
+    };
+
+    #[test]
+    fn test_variable_mixed_case() -> eyre::Result<()> {
+        let linter = SolidityLinter::new()
+            .with_lints(Some(vec![VARIABLE_MIXED_CASE]))
+            .with_buffer_emitter(true);
+
+        let emitted = linter.lint_file(Path::new("testdata/MixedCase.sol")).unwrap().to_string();
+        let warnings = emitted.matches(&format!("warning: {}", VARIABLE_MIXED_CASE.id())).count();
+        let notes = emitted.matches(&format!("note: {}", VARIABLE_MIXED_CASE.id())).count();
+
+        assert_eq!(warnings, 0, "Expected 0 warnings");
+        assert_eq!(notes, 5, "Expected 5 notes");
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_function_mixed_case() -> eyre::Result<()> {
+        let linter = SolidityLinter::new()
+            .with_lints(Some(vec![FUNCTION_MIXED_CASE]))
+            .with_buffer_emitter(true);
+
+        let emitted = linter.lint_file(Path::new("testdata/MixedCase.sol")).unwrap().to_string();
+        let warnings = emitted.matches(&format!("warning: {}", FUNCTION_MIXED_CASE.id())).count();
+        let notes = emitted.matches(&format!("note: {}", FUNCTION_MIXED_CASE.id())).count();
+
+        assert_eq!(warnings, 0, "Expected 0 warnings");
+        assert_eq!(notes, 3, "Expected 3 notes");
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_screaming_snake_case() -> eyre::Result<()> {
+        let linter = SolidityLinter::new()
+            .with_lints(Some(vec![SCREAMING_SNAKE_CASE]))
+            .with_buffer_emitter(true);
+
+        let emitted =
+            linter.lint_file(Path::new("testdata/ScreamingSnakeCase.sol")).unwrap().to_string();
+        let warnings = emitted.matches(&format!("warning: {}", SCREAMING_SNAKE_CASE.id())).count();
+        let notes = emitted.matches(&format!("note: {}", SCREAMING_SNAKE_CASE.id())).count();
+
+        assert_eq!(warnings, 0, "Expected 0 warnings");
+        assert_eq!(notes, 9, "Expected 9 notes");
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_struct_pascal_case() -> eyre::Result<()> {
```

Comment:
these are ok for now but I would like to import the test runner and logic from solar, which, like rustc, uses a directory walk and runs each file individually with custom annotations for config and expected errors; see https://docs.rs/ui_test/latest/ui_test/ and https://github.com/paradigmxyz/solar/blob/main/CONTRIBUTING.md#tests